packages/python/accern_xyme/util.py

```diff
@@ -573,3 +573,41 @@ def maybe_json_loads(value: str) -> Any:
         return json.loads(escape_str(value))
     except json.JSONDecodeError:
         return None
+
+
+def compute_parallel(
+        tasks: List[int],
+        computing_fn: Callable[[int, threading.RLock], Any],
+        progress_fn: Optional[Callable[..., Any]],
+        max_threads: int) -> List[Any]:
+    a_tasks = list(enumerate(tasks))
+    task_count = 0
+    results = [None] * len(a_tasks)
+    progress_lock = threading.RLock()
+    compute_lock = threading.RLock()
+
+    def runner() -> None:
+        nonlocal task_count
+
+        while True:
+            if len(a_tasks) == 0:
+                break
+            cur_task = a_tasks.pop(0)
```

Comment:
make an explicit default so it doesn't throw

Reply:
you can only work with atomic operations otherwise you need a lock

check against None after:
```
cur_task = a_tasks.pop(0, default=None)
if cur_task is None or exc is not None:
   break
```